src/Microsoft.DocAsCode.Common/FileAbstractLayer/FolderRedirectionManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System;
+    using System.Collections.Generic;
+
+    public class FolderRedirectionManager
+    {
+        private Dictionary<string, RelativePath> _redirections = new Dictionary<string, RelativePath>();
+
+        public void AddFolderRedirectionRule(string from, string to)
```

Comment:
>AddFolderRedirectionRule [](start = 20, length = 24)

Use .ctor to construct the redirection rules instead of exposing `Add` method, so that the redirection rules are immutable and thread safety issue can be simplified?